superdwarf-lua.lua

```diff
@@ -64,69 +46,59 @@ function AddSuperdwarf(unit)
 end
 
 function DeleteSuperdwarf(unit)
-    local index = 0
-    for k,v in pairs(superdwarfIds) do
-        if v == unit.id then
-            index = k
-            break
-        end
+    if superdwarfIds[unit.id] then
+        superdwarfIds[unit.id] = nil
+    else
+        qerror("Dwarf was not already super")
     end
-    table.remove(superdwarfIds, index)
+    superdwarfIds[unit.id] = nil
 end
 
 function ListSuperdwarfs()
     print("Current Superdwarfs: ")
-    for k,v in pairs(superdwarfIds) do
-        print(df.unit.find(v).name)
+    for k,_ in pairs(superdwarfIds) do
+        print("[" .. k .. "] " .. dfhack.TranslateName(df.unit.find(k).name))
     end
 end
 
 function ClearSuperdwarfs()
-    local count = #superdwarfIds
-    for i=0, count do superdwarfIds[i]=nil end
+    for k,_ in pairs(superdwarfIds) do superdwarfIds[k] = nil end
 end
 
-function main(...)
-    if args.help then 
+function main(argCommand)
+    if argCommand == 'help' or not argCommand then 
         print(help)
         return
     end
 
-    if args.clear then
+    if argCommand == 'clear' then
         ClearSuperdwarfs()
+        print("Cleared Superdwarfs")
         return
     end
 
-    if args.list then
+    if argCommand == 'list' then
         ListSuperdwarfs()
         return
     end
 
-    unit = nil
-
-    if args.unit then
-        local id = tonumber(args.unit)
-        if id then
-            unit = df.unit.find(id)
-        else
-            qerror("Invalid ID Provided")
-        end
-    else
-        unit = dfhack.gui.getSelectedUnit()
-    end
-
+    local unit = dfhack.gui.getSelectedUnit()
     if not unit then
         qerror("Invalid unit selection")
```

Comment:
Minor thing: both of these lines will print an error message if no unit is selected. I would recommend getting rid of the `qerror()` line to reduce it to just one message. You could instead silence the first one by using `getSelectedUnit(true)`, but I think the error from getSelectedUnit is a bit more clear (and standardized).

Reply:
Was unsure if a `dfhack.gui.getSelectedUnit()` call would block if it failed like that, good to know.